src/audio/module_adapter/module_adapter.c

```diff
@@ -560,10 +670,15 @@ static void module_adapter_process_output(struct comp_dev *dev)
 	struct list_item *blist;
 	int i = 0;
 
+	/* If module has input only just return */
```

Comment:
I think what you're missing in your component implementations is the simple_copy flag. This function would really be used only component do not produce period bytes in every period